app/core/explorer.js

```diff
@@ -1,77 +1,41 @@
 // @flow
 import { openExternal } from './electron'
-import { EXPLORERS } from '../core/constants'
-import { isMainNetwork } from '../core/networks'
-
-export const getExplorerBaseURL = (networkId: string, explorer: ExplorerType) => {
-  let baseURL
-  if (explorer === EXPLORERS.NEO_TRACKER) {
-    if (isMainNetwork(networkId)) {
-      baseURL = 'https://neotracker.io'
-    } else {
-      baseURL = 'https://testnet.neotracker.io'
-    }
-  } else if (explorer === EXPLORERS.NEO_SCAN) {
-    if (isMainNetwork(networkId)) {
-      baseURL = 'https://neoscan.io'
-    } else {
-      baseURL = 'https://neoscan-testnet.io'
-    }
-  } else {
-    if (isMainNetwork(networkId)) {
-      baseURL = 'http://antcha.in'
-    } else {
-      baseURL = 'http://testnet.antcha.in'
-    }
+import explorerConfig from './explorerConfig'
+import { isMainNetwork } from './networks'
+
+export const buildExplorerLink = (
+  networkId: string,
+  explorer: ExplorerType,
+  structure: string,
+  suffix: string
+) => {
+  if (Object.prototype.hasOwnProperty.call(explorerConfig, explorer)) {
+    const explorerInfo = explorerConfig[explorer]
+
+    return getExplorerBaseURL(networkId, explorer) + explorerInfo[structure] + suffix
   }
-  return baseURL
-}
 
-export const getExplorerTxLink = (networkId: string, explorer: ExplorerType, txId: string) => {
-  const baseURL = getExplorerBaseURL(networkId, explorer)
-
-  if (explorer === EXPLORERS.NEO_TRACKER) {
-    return `${baseURL}/tx/${txId}`
-  } else if (explorer === EXPLORERS.NEO_SCAN) {
-    return `${baseURL}/transaction/${txId}`
-  } else {
-    return `${baseURL}/tx/hash/${txId}`
-  }
+  return ''
```

Comment:
better throw an error... we want to know about this

Reply:
Is there an example of an error you can provide?

Reply:
maybe something like this:

`throw new Error(`${explorer} is not a valid explorer type. Valid explorer types are: ${Object.keys(explorerConfig).join(', ')}`)`